core/src/main/java/hudson/model/Job.java

```diff
@@ -221,29 +221,20 @@ public void onLoad(ItemGroup<? extends Item> parent, String name)
 
         TextFile f = getNextBuildNumberFile();
         if (f.exists()) {
-            // starting 1.28, we store nextBuildNumber in a separate file.
-            // but old Hudson didn't do it, so if the file doesn't exist,
-            // assume that nextBuildNumber was read from config.xml
             try {
                 synchronized (this) {
                     this.nextBuildNumber = Integer.parseInt(f.readTrim());
                 }
             } catch (NumberFormatException e) {
                 LOGGER.log(Level.WARNING, "Corruption in {0}: {1}", new Object[] {f, e});
-                //noinspection StatementWithEmptyBody
-                if (this instanceof LazyBuildMixIn.LazyLoadingJob) {
-                    // allow LazyBuildMixIn.onLoad to fix it
```

Comment:
It no longer does so, so simplify and use the same logic for all `Job` types. (In practice all jobs in common usage use `LazyBuildMixIn`.)